src/config.js

```diff
@@ -42,26 +42,29 @@ module.exports = (store) => {
      * @returns {void}
      */
     set (key, value) {
-      if (!key || typeof key !== 'string') {
-        throw new Error('Invalid key type')
+      if (arguments.length === 1) {
+        value = key
+        key = undefined
+      } else if (!key || typeof key !== 'string') {
+        throw new Error('Invalid key type: ' + typeof key)
       }
 
       if (value === undefined || Buffer.isBuffer(value)) {
-        throw new Error('Invalid value type')
+        throw new Error('Invalid value type: ' + typeof value)
       }
 
-      setQueue.push({
+      return setQueue.add(() => _doSet({
```

Comment:
My refactor was to actually just remove the queue/worker, it seemed overkill and not utilized especially because the worker had a concurrency of 1. Additionally, it seems like the concurrent setting of config could be handled by the consuming application more cleanly. Keeping like for like, though +1 for the p-queue change!